server/utils/comments/comment-history.util.js

```diff
@@ -0,0 +1,32 @@
+import db from '../../models';
+
+const { Comment, CommentHistory } = db;
+
+/**
+ * Insert current comment into CommentHistories table
+ * @param {int} commentId
+ * @returns {Boolean} true/false
+ */
+const saveCommentHistory = async (commentId) => {
+  let success;
+  try {
+    const currentCommit = await Comment.findByPk(commentId);
+    if (currentCommit !== null) {
+      const { comment } = currentCommit;
+      const commentHistory = await CommentHistory.create({
+        comment,
+        commentId,
+      });
+      success = true;
+      return success;
+    }
+    // if we get here, comment doesn't exist
+    success = false;
+    return success;
+  } catch (error) {
+    success = false;
+    return success;
+  }
+};
+
```

Comment:
Please i don't understand the flow of this functionality. Under what circumstance do we call the `saveCommentHistory` function?

Reply:
Same here please @cupidcoder

Reply:
> Please i don't understand the flow of this functionality. Under what circumstance do we call the `saveCommentHistory` function?

@TheDrizzyWay the `saveCommentHistory` would be called immediately before a comment is updated. It saves the current comment into the `CommentHistories` table alongside the `commentId`. @BukkyOmo worked on the `edit comment` endpoint. So this function will be called there.

Reply:
> Same here please @cupidcoder

@Chiazokam there is already an `EOL` space here

Reply:
@cupidcoder Got it. Thanks